frontend/src/app/reports/reports-chart.service.ts

```diff
@@ -0,0 +1,154 @@
+import {Injectable} from '@angular/core';
+import * as c3 from 'c3';
+import {ChartAPI} from 'c3';
+
+@Injectable()
+export class ReportsChartService {
+
+
+    chart: ChartAPI;
+
+    constructor() {
+    }
+
+    /**
+     * a function to make a determination about which type of chart to make and calls helper to convert
+     * json to chart data
+     */
+    makeBarChart(id, chart: string, grouping: string, data: Object) {
+        if (grouping.match('quarter')) {
+            this.makeBarChartNumObsByQuarter(id, data);
+        } else if (grouping.match('month')) {
+            this.makeBarChartNumObsByMonth(id, data);
+        } else if (grouping.match('week')) {
+            this.makeBarChartNumObsByWeek(id, data);
+        } else {
+            this.makeBarChartWithoutSubgrouping(id, data);
+        }
+    }
+
+    /**
+     * converts json for number of observations by a single grouping (e.g. year, observer, etc. )
+     */
+    makeBarChartWithoutSubgrouping(id, data: Object) {
+        const names: string[] = new Array();
+        const values: [[string | number]] = [['Count']];
+
+        for (const key of Object.keys(data)) {
+            names.push(key);
+            values[0].push(data[key]);
+        }
+        let number = 0;
+        if (names.length > 5) {
+            number = -70;
```

Comment:
`number` should be renamed to show intent. It looks like it has something to do with rotation. So maybe `degrees`?